x-pack/solutions/search/plugins/search_query_rules/public/components/query_ruleset_detail/query_rule_draggable_list.tsx

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import {
+  EuiDragDropContext,
+  EuiDroppable,
+  EuiDraggable,
+  EuiPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiText,
+  EuiBadge,
+  EuiTextColor,
+  EuiButtonIcon,
+  useEuiTheme,
+  euiDragDropReorder,
+} from '@elastic/eui';
+import {
+  QueryRulesQueryRule,
+  QueryRulesQueryRuleCriteria,
+} from '@elastic/elasticsearch/lib/api/types';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+
+export interface QueryRuleDraggableListProps {
+  rules: QueryRulesQueryRule[];
+  onReorder: (queryRules: QueryRulesQueryRule[]) => void;
+}
+export const QueryRuleDraggableList: React.FC<QueryRuleDraggableListProps> = ({
+  rules,
+  onReorder,
+}) => {
+  const {
+    euiTheme: { colors, size, base },
+  } = useEuiTheme();
+  const DOC_COUNT_COLUMN_SIZING = {
+    columnMinWidth: base * 10,
+    actionMinWidth: base * 5,
+    docCountMinWidth: base * 4,
+    docTextMinWidth: size.l,
+    headerpaddingLeft: base * 3.5,
+  };
+  const criteriaDisplay = (criteria: QueryRulesQueryRuleCriteria) => (
+    <EuiText size="s">
+      <EuiBadge>{criteria.metadata}</EuiBadge>&nbsp;
+      <EuiTextColor color={colors.textPrimary}>{criteria.type}</EuiTextColor>&nbsp;
+      {criteria.values?.join(', ')}
+    </EuiText>
+  );
+  const actionIcon = (queryRule: QueryRulesQueryRule) => (
+    <EuiFlexGroup
+      responsive={false}
+      alignItems="center"
+      gutterSize="m"
+      justifyContent="center"
+      css={{ minWidth: DOC_COUNT_COLUMN_SIZING.columnMinWidth }}
```

Comment:
nitpick: should we put these in a `styles.ts` using emotion? [example I've been trying](https://github.com/elastic/kibana/blob/main/x-pack/solutions/search/plugins/search_playground/public/components/query_mode/styles.ts)

the individual css objects seems like a code smell to me, but we also don't have consistency on when to do this vs grouping them so idk 🤷